torchgeo/trainers/classification.py

```diff
@@ -214,12 +214,17 @@ def validation_step(
             and hasattr(self.logger, 'experiment')
             and hasattr(self.logger.experiment, 'add_figure')
         ):
-            datamodule = self.trainer.datamodule
-            batch['prediction'] = y_hat.argmax(dim=-1)
+            match self.hparams['task']:
+                case 'binary' | 'multilabel':
+                    batch['prediction'] = (y_hat >= 0.5).long()
```

Comment:
0.5 is also the threshold used by torchmetrics

Reply:
https://lightning.ai/docs/torchmetrics/stable/classification/accuracy.html#binaryaccuracy torchmetrics does this automatically, if you pass a float

Reply:
Yep, this part is for plotting, not for metrics.